config/jobs/cert-manager/cert-manager-presubmits.yaml

```diff
@@ -524,47 +524,44 @@ presubmits:
         - name: ndots
           value: "1"
 
-  # An E2E test job to allow us to manually trigger the Venafi TPP E2E tests
-  # with the following GitHub comment:
-  #
-  #  /test pull-cert-manager-e2e-v1-21-feature-issuers-venafi-tpp
-  #
-  # See https://github.com/jetstack/cert-manager/issues/3555
-  #
-  - name: pull-cert-manager-e2e-v1-21-feature-issuers-venafi-tpp
-    always_run: false
-    optional: true
+  # Verifies upgrade from the latest published release with both Helm chart and
+  # static manifests. This is an optional test.
+  - name: pull-cert-manager-upgrade-v1-21
+    # Run only when requested.
+    always_run: true
+    optional: false
+    # No more than 4 instances of this job at the same time.
     max_concurrency: 4
+    # This job will run on Kubernetes cluster.
     agent: kubernetes
+    # Pod utilities will be set up.
     decorate: true
-    branches: []
+    branches:
+      - master
+      - release-1.5
     annotations:
-      description: Runs the E2E tests labelled [Feature:Issuers:Venafi:TPP] against a Kubernetes v1.21 cluster
+      description: Runs cert-manager upgrade from latest published release
     labels:
       preset-service-account: "true"
       preset-dind-enabled: "true"
       preset-bazel-remote-cache-enabled: "true"
       preset-bazel-scratch-dir: "true"
-      preset-venafi-tpp-credentials: "true"
-      preset-retry-flakey-tests: "true"
     spec:
       containers:
       - image: eu.gcr.io/jetstack-build-infra-images/bazelbuild:20210323-ad5071a-3.7.2
         args:
         - runner
-        - devel/ci-run-e2e.sh
-        - -ginkgo.focus
-        - '\[Feature:Issuers:Venafi:TPP\]'
+        - make
+        - cluster
+        - verify_upgrade
         resources:
           requests:
             cpu: 3500m
             memory: 12Gi
```

Comment:
note: I didn't realize our e2e needed so much memory 🤯

Reply:
I think we don't strictly need that much (or maybe we do, I haven't checked how much we consume).
We use n1-standard-8 machines for workers- the 3500m cpu means 2 jobs can fit on a node which also has 32 GB of memory so allocating 12GB to each kind of makes sense, as you wouldn't fit more jobs onto the same node anyway.

Maybe worth exploring if we could use machines with more CPU vs memory.